storybookjs/storybook
# Svelte: use `SvelteComponent` instead of deprecated `SvelteComponentTyped`

## What I did

Replaced deprecated `SvelteComponentTyped` with the `SvelteComponent` type. `SvelteComponentTyped` is deprecated since Svelte 4 (see the [migration guide](https://svelte.dev/docs/v4-migration-guide#sveltecomponenttyped-is-deprecated)) and it is recommended to use the `SvelteComponent` type instead.

The `@storybook/svelte` package marks Svelte 4 as lowest supported version, so it feels reasonable to migrate. Reference:
https://github.com/storybookjs/storybook/blob/a606619e6cc628c083ff1fe5c11d34e3a518ce0f/code/renderers/svelte/package.json#L71-L73

## Checklist for Contributors

### Testing

#### The changes in this PR are covered in the following automated tests:

#### Manual testing

The change is in type test file. If I got it right, `svelte-check` is used to type check the file. Running `svelte-check` does not report errors.

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
